### ISSUE TITLE
Giving sndcpy.apk PROJECT_MEDIA permission to bypass the prompt to allow screen recording.

### ISSUE THREAD

**Issue by Practicalbutterfly5** [rom1v/sndcpy#180]
`adb shell appops set com.rom1v.sndcpy PROJECT_MEDIA allow` can be used to bypass the `Start Recording` dialog. Another thing I noticed with it was that, I needed to insert a `sleep 1` command before the last vlc command so that the MainActivity gets enough time to start.

Can this be implemented?

**Comment by rom1v** [comment]
Oh, great :+1:

I implemented it on `master`: https://github.com/rom1v/sndcpy/commit/026326b36524d3f4af010519f3ca9126dc264bfc

Please test :wink:

Even if I did not plan to do any other release of sndcpy (it's just a PoC), I'll do an exception for this I think :)

**Comment by Practicalbutterfly5** [comment]
I used android studio to compile the sndcpy apk. However on compiling it with `targetSdkVersion 31` and `compileSdkVersion 31` the app crashes when MainActivity is launched on android 12. It works fine on android 10. Here is the logcat.
<details><summary>LOGCAT android 12</summary>
<p>

```
2021-11-23 06:35:47.473 10188-10188/? E/libprocessgroup: Failed to write '10188' to /sys/fs/cgroup/uid_10321/pid_10188/cgroup.procs: Permission denied
2021-11-23 06:35:47.473 10188-10188/? E/Zygote: createProcessGroup(10321, 0) failed: Permission denied
2021-11-23 06:35:47.469 10188-10188/? W/main: type=1400 audit(0.0:7142): avc: denied { write } for dev="cgroup2" ino=2 scontext=u:r:zygote:s0 tcontext=u:object_r:unlabeled:s0 tclass=file permissive=0
2021-11-23 06:35:47.479 10188-10188/? I/om.rom1v.sndcp: Late-enabling -Xcheck:jni
2021-11-23 06:35:47.542 10188-10188/? W/om.rom1v.sndcp: Core platform API violation: Ljava/lang/reflect/Field;->accessFlags:I from Landroid/os/Build; using reflection
2021-11-23 06:35:47.546 10188-10188/? D/ProcessState: Binder ioctl to enable oneway spam detection failed: Invalid argument
2021-11-23 06:35:47.600 10188-10188/? D/CompatibilityChangeReporter: Compat change id reported: 171979766; UID 10321; state: ENABLED
2021-11-23 06:35:47.645 10188-10188/? D/NetworkSecurityConfig: No Network Security Config specified, using platform default
2021-11-23 06:35:47.646 10188-10188/? D/NetworkSecurityConfig: No Network Security Config specified, using platform default
2021-11-23 06:35:47.647 10188-10188/? D/CompatibilityChangeReporter: Compat change id reported: 1531297613045645771; UID 10321; state: DISABLED
2021-11-23 06:35:47.648 10188-10188/? D/CompatibilityChangeReporter: Compat change id reported: 7528921493777479941; UID 10321; state: DISABLED
2021-11-23 06:35:47.659 10188-10188/? D/CompatibilityChangeReporter: Compat change id reported: 160794467; UID 10321; state: ENABLED
2021-11-23 06:35:47.660 10188-10188/? D/AndroidRuntime: Shutting down VM
2021-11-23 06:35:47.662 10188-10188/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.rom1v.sndcpy, PID: 10188
    java.lang.RuntimeException: Unable to create service com.rom1v.sndcpy.RecordService: java.lang.IllegalArgumentException: com.rom1v.sndcpy: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:4502)
        at android.app.ActivityThread.access$1700(ActivityThread.java:248)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2073)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7841)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:550)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.IllegalArgumentException: com.rom1v.sndcpy: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
        at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
        at android.app.PendingIntent.buildServicePendingIntent(PendingIntent.java:724)
        at android.app.PendingIntent.getService(PendingIntent.java:686)
        at com.rom1v.sndcpy.RecordService.createStopAction(RecordService.java:118)
        at com.rom1v.sndcpy.RecordService.createNotification(RecordService.java:105)
        at com.rom1v.sndcpy.RecordService.onCreate(RecordService.java:62)
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:4489)
        	... 9 more

```
</p>
</details>

Fixed by #181
